### PR TITLE
xml: add windows-eventlog-xml-parser() 

### PR DESCRIPTION
**Pull request by alltilla**
Signed-off-by: Attila Szakacs <attila.szakacs@axoflow.com>

### CODE REVIEW

**Review by MrAnno — APPROVED** [review]
:)

File: modules/xml/tests/test_windows_eventlog_parser.c [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  cr_assert_eq(type, LM_VT_STRING);
+  cr_assert_str_eq(value, "bar");
+
+  value = log_msg_get_value_by_name_with_type(msg, ".winlog.Event.EventData.Data", &value_len, &type);
+  cr_assert_eq(type, LM_VT_NULL);
+  cr_assert_str_eq(value, "");
+
+  value = log_msg_get_value_by_name_with_type(msg, ".winlog.Event.EventData.Data._Name", &value_len, &type);
+  cr_assert_eq(type, LM_VT_NULL);
+  cr_assert_str_eq(value, "");
```
Comment (MrAnno) [inline modules/xml/tests/test_windows_eventlog_parser.c:112]: Optional: can we add an assert to check that the non-Data part of the parsing remains correct? For example, we could assert on a RenderingInfo attribute or on the EventID.

File: modules/xml/xml-plugin.c [hunk 2]
```diff
@@ -33,6 +33,11 @@ static Plugin xml_plugins[] =
     .name = "xml",
     .parser = &xml_parser,
   },
+  {
+    .type = LL_CONTEXT_PARSER,
+    .name = "windows-eventlog-parser",
+    .parser = &xml_parser,
```
Comment (bazsi) [inline modules/xml/xml-plugin.c:39]: just noticed that we have an SCL with the same name.

scl/windowseventlog/plugin.conf

**Review by bazsi — COMMENTED** [review]
Apart from the name clash this looks good to me.

File: modules/xml/tests/test_windows_eventlog_parser.c (hunk 1 above)
Comment (alltilla) [reply modules/xml/tests/test_windows_eventlog_parser.c]: Yes, implemented :) Thanks!

File: modules/xml/xml-plugin.c (hunk 2 above)
Comment (alltilla) [reply modules/xml/xml-plugin.c:39]: Good catch! Renamed it to `windows-eventlog-xml-parser()`.

**Review by bazsi — APPROVED** [review]

**Review by MrAnno — APPROVED** [review]

**Pull request merged**